presto-main/src/main/java/com/facebook/presto/operator/scalar/MapIndeterminateOperator.java

```diff
@@ -0,0 +1,69 @@
+package com.facebook.presto.operator.scalar;
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.StandardErrorCode;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.SqlType;
+import com.google.common.base.Throwables;
+
+import javax.annotation.Nullable;
+
+import java.lang.invoke.MethodHandle;
+
+import static com.facebook.presto.metadata.OperatorType.INDETERMINATE;
+import static com.facebook.presto.spi.type.TypeUtils.readNativeValue;
+
+@ScalarOperator(INDETERMINATE)
+public final class MapIndeterminateOperator
+{
+    private MapIndeterminateOperator() {}
+
+    @TypeParameter("K")
+    @TypeParameter("V")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean indeterminate(
+            @OperatorDependency(operator = INDETERMINATE, returnType = StandardTypes.BOOLEAN, argumentTypes = {"K"}) MethodHandle keyFunction,
```

Comment:
Interestingly, a map with indeterminate keys doesn't make much sense. There's no way to look them up because they are not equal to anything.